dot/services_test.go

```diff
@@ -0,0 +1,656 @@
+	}{
+		{
+			name:      "nil BlockState test",
+			service:   &state.Service{},
+			expectNil: true,
+			err:       errors.New("cannot have nil EpochState"),
+		},
+		{
+			name:      "working example",
+			service:   stateSrvc,
+			expectNil: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := nodeInstance.createBlockVerifier(tt.service)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+
+			if tt.expectNil {
+				assert.Nil(t, got)
+			} else {
+				assert.NotNil(t, got)
+			}
+		})
+	}
+}
+
+func Test_createRuntimeStorage(t *testing.T) {
+	cfg := NewTestConfig(t)
+
+	genFile := newTestGenesisRawFile(t, cfg)
+
+	cfg.Init.Genesis = genFile
+
+	nodeInstance := nodeBuilder{}
+	err := nodeInstance.initNode(cfg)
+	require.NoError(t, err)
+
+	stateSrvc, err := nodeInstance.createStateService(cfg)
+	require.NoError(t, err)
+
+	tests := []struct {
+		name           string
+		service        *state.Service
+		expectedBaseDB *state.BaseState
+		err            error
+	}{
+		{
+			name:           "working example",
+			service:        stateSrvc,
+			expectedBaseDB: stateSrvc.Base,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := nodeInstance.createRuntimeStorage(tt.service)
+			assert.ErrorIs(t, err, tt.err)
+			assert.Equal(t, tt.expectedBaseDB, got.BaseDB)
+			assert.NotNil(t, got.LocalStorage)
+			assert.NotNil(t, got.PersistentStorage)
+			assert.IsType(t, &runtime.NodeStorage{}, got)
+		})
+	}
+}
+
+func Test_createSystemService(t *testing.T) {
+	cfg := NewTestConfig(t)
+
+	genFile := newTestGenesisRawFile(t, cfg)
+
+	cfg.Init.Genesis = genFile
+
+	nodeInstance := nodeBuilder{}
+	err := nodeInstance.initNode(cfg)
+	require.NoError(t, err)
+
+	stateSrvc, err := nodeInstance.createStateService(cfg)
+	require.NoError(t, err)
+
+	type args struct {
+		cfg     *types.SystemInfo
+		service *state.Service
+	}
+	tests := []struct {
+		name      string
+		args      args
+		expectNil bool
+		err       error
+	}{
+		{
+			name: "working example",
+			args: args{
+				service: stateSrvc,
+			},
+			expectNil: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := nodeInstance.createSystemService(tt.args.cfg, tt.args.service)
+			assert.ErrorIs(t, err, tt.err)
+
+			// TODO: change this check to assert.Equal after state.Service interface is implemented.
+			if tt.expectNil {
+				assert.Nil(t, got)
+			} else {
+				assert.NotNil(t, got)
+				assert.IsType(t, &system.Service{}, got)
+			}
+		})
+	}
+}
+
+func Test_newInMemoryDB(t *testing.T) {
+	tests := []struct {
+		name      string
+		path      string
+		expectNil bool
+		err       error
+	}{
+		{
+			name:      "working example",
+			path:      "test_data",
+			expectNil: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := newInMemoryDB(tt.path)
```

Comment:
nit A bit out of scope, but why does it need a file path if it's in-memory?

Reply:
Good point, I've removed the path parameter.